src/wrapper/strings/java_str.rs

```diff
@@ -27,11 +28,74 @@ impl<'a: 'b, 'b> JavaStr<'a, 'b> {
         Ok(java_str)
     }
 
-    /// Extract the raw C string pointer from the JavaStr. This will be
+    /// Get a pointer to the character array beneath a [JString]
+    /// The array contains Java's modified UTF-8: `CESU-8`
+    ///
+    /// # Warning
+    ///
+    /// The caller must release the array when they are done with it
+    // This function does not need to be unsafe,
+    // as a memory leak isn't considered unsafe nor is getting a raw pointer to something
+    fn get_string_utf_chars(obj: JString<'_>, env: &JNIEnv<'_>) -> Result<*const c_char> {
+        non_null!(obj, "get_string_utf_chars obj argument");
+        let ptr: *const c_char = jni_non_null_call!(
+            env.internal,
+            GetStringUTFChars,
+            obj.into_raw(),
+            ::std::ptr::null::<jboolean>() as *mut jboolean
+        );
+
+        Ok(ptr)
+    }
+
+    /// Get the raw C string pointer from the JavaStr. This will be
     /// encoded using the JVM internal `CESU-8`-style.
     pub fn get_raw(&self) -> *const c_char {
         self.internal
     }
+
+    /// Get the raw C string pointer from the [JavaStr]. This will be
+    /// encoded using the JVM internal `CESU-8`-style.
+    ///
+    /// # Warning
+    /// The programmer is responsible for making sure the backing string gets released
+    /// when they are done with it by resconstructing the [JavaStr] with [Self::from_raw],
+    /// which will release the backing string when it is dropped.
```

Comment:
I wonder if we could borrow more from the similar `Box::into_raw` docs here:

Docs for for `Box::into_raw` say:
> After calling this function, the caller is responsible for the memory previously managed by the Box. In particular, the caller should properly destroy T and release the memory, taking into account the [memory layout](https://doc.rust-lang.org/std/boxed/index.html#memory-layout) used by Box. The easiest way to do this is to convert the raw pointer back into a Box with the [Box::from_raw](https://doc.rust-lang.org/std/boxed/struct.Box.html#method.from_raw) function, allowing the Box destructor to perform the cleanup.

In particular it would be good to highlight that `from_raw` would be the recommended way to handle freeing the raw pointer later.

Reply:
Good point. Will fix

Reply:
oh, sorry was rushing here - the docs do actually highlight `from_raw` already

Reply:
Oops, accidentally unresolved it 😅